File: mgl-webapp/src/main/java/com/organization/mvcproject/controller/GameController.java
package com.organization.mvcproject.controller;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.organization.mvcproject.api.service.GameService;
import com.organization.mvcproject.model.GameImpl;


@RequestMapping(value = "/game")
@RestController
public class GameController {

	@Autowired
	private GameService gameService;

	@GetMapping(value = "/getAll")
	public ResponseEntity<List<GameImpl>> fetchAllGames() {
		return new ResponseEntity<List<GameImpl>>(gameService.retrieveAllGames(), HttpStatus.OK);
	}

	@PostMapping(value = "/", consumes = MediaType.APPLICATION_JSON_VALUE)
	public ResponseEntity<Void> createGame(@RequestBody GameImpl game) {
		gameService.saveGame(game);
		return new ResponseEntity<Void>(HttpStatus.CREATED);
	}
	
	@PutMapping(value = "/", consumes = MediaType.APPLICATION_JSON_VALUE)
	public ResponseEntity<?> updateGame(@RequestBody GameImpl game){
		GameImpl savedGame = (GameImpl) gameService.saveGame(game);
		return new ResponseEntity<>(savedGame, HttpStatus.OK);
	}
	@DeleteMapping(value = "/{id}")
	public ResponseEntity<?> deleteGame(@PathVariable("id") String id) {
		return new ResponseEntity<>(gameService.deleteGame(Long.valueOf(id)), HttpStatus.OK);
	}
	
	@GetMapping(value = "/{id}")
	public ResponseEntity<?> findGameById(@PathVariable("id") String id) {
		return new ResponseEntity<>((GameImpl) gameService.findGameById(Long.valueOf(id)), HttpStatus.OK);
	}
	
}

File: mgl-webapp/src/main/webapp/resources/static/js/GameController.js
'use strict';

angular.module('GameApp').controller('GameController',
		[ 'GameService', function(GameService) {
			var vm = this;
			vm.game = {
				game_id : '',
				game_name : '',
				game_genre : ''
			};
			vm.games = [];

			vm.fetchAllGames = function(){
				GameService.fetchAllGames().then(function(data) {
					vm.games = data;
				}).finally(function(){
					vm.game = {};
				});
			}

			vm.addGame = function(){
				return GameService.createGame(vm.game).then( function() {
					vm.fetchAllGames();
				});
			}
			
			vm.loadGame = function(gameToUpdate){
				vm.game = angular.copy(gameToUpdate);
			}
			
			vm.updateGame = function(){
				return GameService.updateGame(vm.game).then( function() {
					vm.fetchAllGames();
				});
			}
			
			vm.deleteGame = function(gameId){
				return GameService.deleteGame(gameId).then( function() {
					vm.fetchAllGames();
				});
			}
			
			vm.fetchAllGames();
		} ]);
